### PR TITLE
[fix]: correctly retrieve the fileSize on 32 bits OS

### PR DESCRIPTION
**Pull request by adoussot**
"size_t" size depends on the OS. On 32-bit system it can only hold a 32
bits variable = 4 294 967 296, which is not enough to store a big file
size (e.g: 14Go). We should use a 64-bits variable

### CODE REVIEW

File: src/AvTranscoder/properties/FileProperties.cpp [hunk 1]
```diff
@@ -210,7 +210,7 @@ size_t FileProperties::getBitRate() const
     return _avFormatContext->bit_rate;
 }
 
-size_t FileProperties::getFileSize() const
+unsigned __int64 FileProperties::getFileSize() const
```
Comment (MarcAntoine-Arnaud) [inline src/AvTranscoder/properties/FileProperties.cpp]: is it possible to define more something like in TuttleOFX:
https://github.com/tuttleofx/TuttleOFX/blob/develop/libraries/tuttle/src/tuttle/common/system/compatibility.hpp

**Review by MarcAntoine-Arnaud — CHANGES_REQUESTED** [review]

File: src/AvTranscoder/properties/FileProperties.cpp (hunk 1 above)
Comment (adoussot) [reply src/AvTranscoder/properties/FileProperties.cpp]: Do you want me to define a new type in std namespace ?

e.g (this is pseudo-code)

```c++
// compatibility problems...
namespace std
{
#ifdef 32bit-OS // use https://stackoverflow.com/questions/1505582/determining-32-vs-64-bit-in-c to define the system type
	typedef Custom_size_t c_size_t // can hold 64-bits
#else 
	typedef ::size_t c_size_t
#endif
}
```

I'm not really use to c++ but doesn't it seem a bit weird ?

Comment (cchampet) [reply src/AvTranscoder/properties/FileProperties.cpp]: LGTM

Comment (fabiencastan) [reply src/AvTranscoder/properties/FileProperties.cpp]: Why not just use `uint64_t`? It is multi-platform (`#include <cstdint>`) and dedicated to provide fixed size integer.

Comment (cchampet) [reply src/AvTranscoder/properties/FileProperties.cpp]: Indeed  __uint64_t__ seems to be what you are looking for (it is provided by the C library).
Did you test it @adoussot ?

Comment (adoussot) [reply src/AvTranscoder/properties/FileProperties.cpp]: Yes, i thought `unsigned __int64` was cross platform (which is not). so `uint64_t` is correct. Thanks!

**Comment by adoussot** [comment]
Is this Ok (RTM) ? Or should we still change something ?

**Comment by fabiencastan** [comment]
Would be good to add a comment to the function , like
```
/**
 * @note Ensure 64 bits size even on 32 bits OS with uint64_t, so we can manage large files > 4Go.
 */
```

**Review by MarcAntoine-Arnaud — APPROVED** [review]

**Comment by adoussot** [comment]
Since the previous PR https://github.com/avTranscoder/avTranscoder/pull/305 was incrementing the avtranscoder version directly in develop (and didn't merge to master) I stick to the same   release policy and incremented micro version in 086d074.

If it's ok for you, could someone merge it ? Without any response, I will merge this PR by the end of the weekend. thanks

**Pull request merged**